svc/api/admin/src/route/mod.rs

```diff
@@ -20,16 +21,40 @@ pub async fn handle(
 
 define_router! {
 	routes: {
-		"cluster" / "server_ips": {
-			GET: cluster::server_ips(
-				query: cluster::ServerIpsQuery,
-			),
-		},
-
 		"login": {
 			POST: login::login(
 				body: models::AdminLoginRequest,
 			),
 		},
+
+		"clusters" / "server_ips": {
+			GET: clusters::server_ips(
+				query: clusters::ServerIpsQuery,
+			),
+		},
+
+		"clusters": {
+			POST: clusters::create(
+				body: models::AdminClustersCreateRequest,
+			),
+		},
+
+		"clusters" / "list": {
+			GET: clusters::get(),
+		},
+
+		"clusters" / Uuid / "datacenters": {
+			POST: clusters::datacenters::create(
+				body: models::AdminClustersDatacentersCreateRequest,
+			),
+		},
+
+		"clusters" / Uuid / "datacenters" / "list": {
```

Comment:
remove list. just get the root to list.

```suggestion
		"clusters" / Uuid / "datacenters": {
```